dianna/methods/kernelshap.py

```diff
@@ -204,13 +206,13 @@ def _runner(self, features):
             features (np.ndarray): A matrix of samples (# samples x # features)
                                    on which to explain the model's output.
         """
-        return prepare(self.onnx_model).run(
-            self._mask_image(
-                features,
-                self.image_segments,
-                self.input_data,
-                self.background,
-                self.channels_axis_index,
-                self.input_node_dtype.as_numpy_dtype
-            )
-        )[f"{self.output_node}"]
+        model_input = self._mask_image(features,
+                                       self.image_segments,
+                                       self.input_data,
+                                       self.background,
+                                       self.channels_axis_index,
+                                       self.input_node_dtype.as_numpy_dtype
+                                       )
+        if self.preprocess_function is not None:
```

Comment:
I wonder if it makes more sense to move the `preprocess_function` before calling the segmentation (line 122). If the user wants to preprocess the input data, I guess it happens before the segmentation and masking. What do you think @loostrum ?